scheduler/ecs/ecs.go

```diff
@@ -288,6 +278,41 @@ func (m *Scheduler) Instances(ctx context.Context, appID string) ([]*scheduler.I
 	return instances, nil
 }
 
+func (m *Scheduler) describeTaskDefinitions(ctx context.Context, tasks []*ecs.Task) (map[string]*ecs.TaskDefinition, error) {
+	taskDefinitions := make(map[string]*ecs.TaskDefinition)
+
+	for _, t := range tasks {
+		k := *t.TaskDefinitionArn
+		taskDefinitions[k] = nil
+	}
+
+	ch := make(chan struct {
+		resp *ecs.DescribeTaskDefinitionOutput
+		err  error
+	}, len(taskDefinitions))
+	for arn := range taskDefinitions {
+		go func(arn string) {
+			resp, err := m.ecs.DescribeTaskDefinition(ctx, &ecs.DescribeTaskDefinitionInput{
+				TaskDefinition: aws.String(arn),
+			})
+			ch <- struct {
+				resp *ecs.DescribeTaskDefinitionOutput
+				err  error
+			}{resp, err}
+		}(arn)
+	}
+
+	for arn := range taskDefinitions {
+		result := <-ch
+		if err := result.err; err != nil {
+			return nil, err
+		}
+		taskDefinitions[arn] = result.resp.TaskDefinition
```

Comment:
Have to use result.resp.TaskDefinition.Arn as the key here. `arn` will be the incorrect value.